pal/pacer/pacer-web.ts

```diff
@@ -106,25 +104,46 @@ export class Pacer {
         if (!this._isPlaying) return;
         this._ctTime(this._stHandle);
         this._stHandle = 0;
+        if (this._timeId) {
+            clearTimeout(this._timeId);
+            this._timeId = 0;
+        }
         this._isPlaying = false;
         this._frameCount = 0;
     }
 
     _handleRAF = (): void => {
-        const elapseTime = performance.now() - this._startTime;
+        if (!this._isPlaying) {
+            return;
+        }
+        const currTime = performance.now();
+        const elapseTime = currTime - this._startTime;
         const elapseFrame = Math.floor(elapseTime / this._frameTime);
         if (elapseFrame < this._frameCount) {
-            this._rAF.call(window, this._handleRAF);
-        } else {
-            this._frameCount++;
-            if (this._callback) {
-                this._callback();
+            const sleepMs = this._frameCount * this._frameTime - elapseTime;
+            if (sleepMs > 16) {
+                this._stHandle = this._rAF.call(window, this._handleRAF);
+                return;
+            } else if (sleepMs > 4) {
+                // Avoid timing to sleepMs because timing has accuracy issues.
+                this._timeId = setTimeout(this._handleRAF, sleepMs - 4);
+                return;
+            } else if (sleepMs > 1) {
+                this._timeId = setTimeout(this._handleRAF, 0);
+                return;
+            } else {
+                // Acceptance tolerance within 1ms.
             }
         }
-        if (performance.now() - this._startTime > FRAME_RESET_TIME) {
-            this._startTime = performance.now();
-            this._frameCount = 0;
+        this._frameCount++;
+        if (this._callback) {
+            this._callback();
+        }
+        if (currTime - this._startTime > FRAME_RESET_TIME) {
+            this._startTime = currTime;
+            this._frameCount = 1;
         }
+        this._timeId = setTimeout(this._handleRAF, 0);
```

Comment:
Why do we need another timer to trigger _handleRAF?
`RAF` means `requestAnimationFrame`, its callback is invoked every frame.

Reply:
If 'elapseFrame < 16ms' calling requestAnimationFrame may add an extra frame of time. This results in occasional 30ms frames